src/Atmos/Model/bc_energy.jl

```diff
@@ -114,3 +114,68 @@ function atmos_energy_normal_boundary_flux_second_order!(
     # we want to prescribe the inward flux
     fluxᵀn.ρe -= bc_energy.fn(state⁻, aux⁻, t)
 end
+
+"""
+    BulkFormulaEnergy(fn) :: EnergyBC
+
+Calculate the net inward energy flux across the boundary.
+The drag coefficient is `C_q = fn_C_q(state, aux, t, normPu_int)`.
+The surface temp is `T_sfc= fn_T_sfc(state, aus, t)`.
+The surface q_tot is `q_tot_sfc = fn_q_tot_sfc(state, aux, t)`.
+with signature `fn(state, aux, t)`, returning the flux (in W m^-2).
+"""
+struct BulkFormulaEnergy{FNX,FNT,FNM} <: EnergyBC
+    fn_C_h::FNX
+    fn_T_sfc::FNT
+    fn_q_tot_sfc::FNM
+end
+function atmos_energy_boundary_state!(
+    nf,
+    bc_energy::BulkFormulaEnergy,
+    atmos,
+    args...,
+) 
+end
+function atmos_energy_normal_boundary_flux_second_order!(
+    nf,
+    bc_energy::BulkFormulaEnergy,
+    atmos,
+    fluxᵀn,
+    n⁻,
+    state_sfc⁻,
+    diff_sfc⁻,
+    hyperdiff_sfc⁻,
+    aux_sfc⁻,
+    state_sfc⁺,
+    diff_sfc⁺,
+    hyperdiff_sfc⁺,
+    aux_sfc⁺,
+    bctype,
+    t,
+    state_int⁻,
+    diff_int⁻,
+    aux_int⁻,
+)
+
+    u_int⁻ = state_int⁻.ρu / state_int⁻.ρ
+    Pu_int⁻ = projection_tangential(atmos, aux_int⁻, u_int⁻)
+    normPu_int⁻ = norm(Pu_int⁻)
+    # NOTE: difference from design docs since normal points outwards
+    C_h = bc_energy.fn_C_h(state_sfc⁻, aux_sfc⁻, t, normPu_int⁻)
+    T_sfc = bc_energy.fn_T_sfc(state_sfc⁻, aux_sfc⁻, t)
+    q_tot_sfc = bc_energy.fn_q_tot_sfc(state_sfc⁻, aux_sfc⁻, t)
+    #TODO: add kinetic energy? also figure out a better way...
+    # MSE at the surface
+    q_pt_sfc = PhasePartition(q_tot_sfc)
+    R_m_sfc = gas_constant_air(atmos.param_set, q_pt_sfc)
+    e_int_sfc = internal_energy(atmos.param_set, T_sfc, q_pt_sfc)
+    h_int_sfc = specific_enthalpy(e_int_sfc, R_m_sfc, T_sfc)
+    e_pot_sfc = gravitational_potential(atmos.orientation, aux_sfc⁻)
+    MSE_sfc = h_int_sfc + e_pot_sfc
```

Comment:
Use the `moist_static_energy` function in thermodynamics. Right now it requires a thermodynamic state. If you cannot define that for the surface, you can overload an explicit function in `Thermodynamics` that takes the variables you have here as input. @charleskawczynski can help.